peloton-frontend/component/home/race/RaceDetail.js

```diff
@@ -9,21 +9,73 @@ import { ridersInfoState } from "../../../state/rider/RiderState";
 import { QueryApi } from "../../../utils/api/QueryApi";
 import LoadingIndicator from "../../../utils/LoadingIndicator";
 import { certificationsState } from "../../../state/certification/CertificationState";
-import { COLOR } from "../../../utils/constants";
+import { COLOR, deviceHeight } from "../../../utils/constants";
 import RaceCertificationImages from "./RaceCertificationImages";
 import RaceDetailInfo from "./RaceDetailInfo";
 import RaceSpec from "./RaceSpec";
 import LinkCopyButton from "./LinkCopyButton";
+import { navigateWithHistory } from "../../../utils/util";
+import { useNavigation } from "@react-navigation/core";
+import { CalculationApi } from "../../../utils/api/CalculationApi";
+import HalfWidthButton from "./HalfWidthButton";
 
 const RaceDetail = ({ route }) => {
   const raceId = route.params.id;
+  const navigation = useNavigation();
   const token = useRecoilValue(memberTokenState);
   const setIsLoading = useSetRecoilState(loadingState);
   const [raceInfo, setRaceInfo] = useRecoilState(raceInfoState);
   const [ridersInfo, setRidersInfo] = useRecoilState(ridersInfoState);
-  const [certificationsInfo, setCertificationsInfo] = useRecoilState(
-    certificationsState
-  );
+  const setCertificationsInfo = useSetRecoilState(certificationsState);
+
+  const calculateRace = async () => {
+    setIsLoading(true);
+    let response;
+    try {
+      response = await CalculationApi.post(token, raceId);
```

Comment:
```suggestion
      const {status} = await CalculationApi.post(token, raceId);
```
post가 response.status가 아닌 response까지만 리턴하고 여기서 status를 받아쓰면 좋겠네요.
그리고 아래 if문도 try 문 안에 넣으면 될 것 같아요